core/src/cats/sql_update.cc

```diff
@@ -161,18 +174,33 @@ bool BareosDb::UpdateJobEndRecord(JobControlRecord* jcr, JobDbRecord* jr)
   JobTDate = ttime;
 
   DbLocker _{this};
-  Mmsg(
-      cmd,
-      "UPDATE Job SET JobStatus='%c',Level='%c',EndTime='%s',"
-      "ClientId=%u,JobBytes=%s,ReadBytes=%s,JobFiles=%u,JobErrors=%u,"
-      "VolSessionId=%u,"
-      "VolSessionTime=%u,PoolId=%u,FileSetId=%u,JobTDate=%s,"
-      "RealEndTime='%s',PriorJobId=%s,HasBase=%u,PurgedFiles=%u WHERE JobId=%s",
-      (char)(jr->JobStatus), (char)(jr->JobLevel), dt, jr->ClientId,
-      edit_uint64(jr->JobBytes, ed1), edit_uint64(jr->ReadBytes, ed4),
-      jr->JobFiles, jr->JobErrors, jr->VolSessionId, jr->VolSessionTime,
-      jr->PoolId, jr->FileSetId, edit_uint64(JobTDate, ed2), rdt, PriorJobId,
-      jr->HasBase, jr->PurgedFiles, edit_int64(jr->JobId, ed3));
+  if (jr->JobFiles > 0) {
```

Comment:
Yes, we need to handle this. When canceling a job, the director tries to receive the `JobFiles` from the FD, but as the connection drops because of the `cancel`, the FD is unable to send the data. So the container that was supposed to receive the JobFiles stays in its initialized state i.e. all zeros, and then the director proceeds to update the database with this empty data, which would cause it to overwrite the data sent from the SD during checkpoints.

And yes, the duplication is ugly, I've changed it.